app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt

```diff
@@ -0,0 +1,593 @@
+                            resources.getString(R.string.pref_key_locale),
+                            resources.getString(R.string.pref_key_secure),
+                            resources.getString(R.string.pref_key_default_browser))
+                    .setSettingsProvider(TelemetrySettingsProvider(context))
+                    .setCollectionEnabled(telemetryEnabled)
+                    .setUploadEnabled(telemetryEnabled)
+
+            val serializer = JSONPingSerializer()
+            val storage = FileTelemetryStorage(configuration, serializer)
+            val client = HttpURLConnectionTelemetryClient()
+            val scheduler = JobSchedulerTelemetryScheduler()
+
+            TelemetryHolder.set(Telemetry(configuration, storage, client, scheduler)
+                    .addPingBuilder(TelemetryCorePingBuilder(configuration))
+                    .addPingBuilder(TelemetryEventPingBuilder(configuration))
+                    .setDefaultSearchProvider(createDefaultSearchProvider(context)))
+        } finally {
+            StrictMode.setThreadPolicy(threadPolicy)
+        }
+    }
+
+    private fun createDefaultSearchProvider(context: Context): DefaultSearchMeasurement.DefaultSearchEngineProvider {
+        return DefaultSearchMeasurement.DefaultSearchEngineProvider {
+            SearchEngineManager.getInstance()
+                    .getDefaultSearchEngine(context)
+                    .identifier
+        }
+    }
+
+    /**
+     * Add the position of the current session and total number of sessions as extras to the event
+     * and return it.
+     */
+    @CheckResult
+    private fun withSessionCounts(event: TelemetryEvent): TelemetryEvent {
+        val sessionManager = SessionManager.getInstance()
+
+        event.extra(Extra.SELECTED, sessionManager.positionOfCurrentSession.toString())
+        event.extra(Extra.TOTAL, sessionManager.numberOfSessions.toString())
+
+        return event
+    }
+
+    @JvmStatic
+    fun startSession() {
+        TelemetryHolder.get().recordSessionStart()
+
+        TelemetryEvent.create(Category.ACTION, Method.FOREGROUND, Object.APP).queue()
+    }
+
+    @JvmStatic
+    fun stopSession() {
+        TelemetryHolder.get().recordSessionEnd()
+
+        TelemetryEvent.create(Category.ACTION, Method.BACKGROUND, Object.APP).queue()
+    }
+
+    @JvmStatic
+    fun stopMainActivity() {
+        TelemetryHolder.get()
+                .queuePing(TelemetryCorePingBuilder.TYPE)
+                .queuePing(TelemetryEventPingBuilder.TYPE)
+                .scheduleUpload()
+    }
+
+    @JvmStatic
+    fun urlBarEvent(isUrl: Boolean, autocompleteResult: AutocompleteResult) {
+        if (isUrl) {
+            TelemetryWrapper.browseEvent(autocompleteResult)
+        } else {
+            TelemetryWrapper.searchEnterEvent()
+        }
+    }
+
+    private fun browseEvent(autocompleteResult: AutocompleteResult) {
```

Comment:
Cool, was looking to see what data was collected here, just size, custom/default, and isEmpty is fine, just Type 2 data.